File: java/src/main/java/org/dilithia/sdk/model/NetworkInfo.java
package org.dilithia.sdk.model;

/**
 * Represents network head information.
 *
 * @param blockHeight the current block height
 * @param blockHash   the current block hash
 * @param chainId     the chain identifier
 */
public record NetworkInfo(long blockHeight, String blockHash, String chainId) {}

File: dotnet/Dilithia.Sdk/Models/NetworkInfo.cs
namespace Dilithia.Sdk.Models;

/// <summary>
/// Network information returned by the node.
/// </summary>
/// <param name="ChainId">The chain identifier.</param>
/// <param name="BlockHeight">The current block height.</param>
/// <param name="BaseFee">The current base fee.</param>
public record NetworkInfo(string ChainId, long BlockHeight, long BaseFee);
